File: App/src/Adaptive.ReactiveTrader.Client/Models/Trade.cs
using System;

namespace Adaptive.ReactiveTrader.Client.Models
{
    internal class Trade : ITrade
    {
        public string CurrencyPair { get; private set; }
        public Direction Direction { get; private set; }
        public long Notional { get; private set; }
        public decimal SpotRate { get; private set; }
        public TradeStatus TradeStatus { get; private set; }
        public DateTime TradeDate { get; private set; }
        public long TradeId { get; private set; }
        public string TraderName { get; private set; }
        public DateTime ValueDate { get; private set; }

        public Trade(string currencyPair, Direction direction, long notional, decimal spotRate, TradeStatus tradeStatus, DateTime tradeDate, long tradeId, string traderName, DateTime valueDate)
        {
            CurrencyPair = currencyPair;
            Direction = direction;
            Notional = notional;
            SpotRate = spotRate;
            TradeStatus = tradeStatus;
            TradeDate = tradeDate;
            TradeId = tradeId;
            TraderName = traderName;
            ValueDate = valueDate;
        }
    }
}

File: App/src/Adaptive.ReactiveTrader.Client.JS/Models/Execution/Trade.ts
 class Trade implements ITrade {
    currencyPair: string;
    direction: Direction;
    notional: number;
    spotRate: number;
    tradeStatus: TradeStatus;
    tradeDate: Date;
    tradeId: number;
    traderName:string;
    valueDate:Date;
    dealtCurrency: string;
    
    constructor(
        currencyPair: string,
        direction: Direction,
        notional: number,
        spotRate: number,
        tradeStatus: TradeStatus,
        tradeDate: Date,
        tradeId: number,
        traderName:string,
        valueDate:Date,
        dealtCurrency: string){
    
        this.currencyPair = currencyPair;
        this.direction = direction;
        this.notional = notional;
        this.spotRate = spotRate;
        this.tradeStatu = tradeStatus;
        this.tradeDate = tradeDate;
        this.tradeId = tradeId;
        this.traderName = traderName;
        this.valueDate = valueDate;
        this.dealtCurrency = dealtCurrency;
    }    
}